pkg/odo/cli/init/params/flags.go

```diff
@@ -0,0 +1,27 @@
+package params
+
+import "github.com/redhat-developer/odo/pkg/odo/util"
+
+// FlagsBuilder is a backend that will extract init parameters from flags passed to the command
+type FlagsBuilder struct {
+	flags map[string]string
+}
+
+func (o *FlagsBuilder) IsAdequate(flags map[string]string) bool {
+	// Save args to use them for building params
+	o.flags = flags
+	return len(flags) > 0
+}
+
+func (o *FlagsBuilder) ParamsBuild() (InitParams, error) {
+	if len(o.flags) == 0 {
+		util.LogErrorAndExit(nil, "IsAdequate must be called and return true before to call ParamsBuild")
```

Comment:
This is an assertion and the message is dedicated to the developer that would introduce a bug. The message should not show up if the program is not buggy

Reply:
Alright. Can you still return the error instead of exiting the program?

Reply:
The common practice is to abort the program as soon as possible when an assertion is raised, for example in c++: https://www.cplusplus.com/reference/cassert/assert/

Reply:
But, anytime you return an error, it is stacked up and passed on to LogErrorAndExit. I think abrupt exit calls might make it difficult to debug.

Reply:
So if I understand correctly, if it is a developer intended error/assertion you simply exit the program and exit, but if it is a user intended error, you return?

Reply:
> But, anytime you return an error, it is stacked up and passed on to LogErrorAndExit. I think abrupt exit calls might make it difficult to debug.

Isn't `LogErrorAndExit` the right thing to do? I don't see an abrupt exit call (`os.Exit`) here. I'm confused what you are referring to, because in my head, abrupt exit call is a call to `os.Exit`.

Reply:
> So if I understand correctly, if it is a developer intended error/assertion you simply exit the program and exit, but if it is a user intended error, you return?

This thread gives examples of assertion vs exception (in Go, returning the err is equivalent to an exception): https://stackoverflow.com/questions/1276308/exception-vs-assertion

For an assertion, the code exits immediately because the developer used the module in a manner that is not supported by the module, and you cannot know what will happen in this case (you don't even know, if you return an error, if it will be bubbled up correctly, for example you could have a division by zero just after, and that would be difficult to find the root cause).